oauth2/src/main/java/com/okta/spring/boot/oauth/OktaOAuth2ResourceServerAutoConfig.java

```diff
@@ -45,15 +53,35 @@
     JwtDecoder jwtDecoder(OAuth2ResourceServerProperties oAuth2ResourceServerProperties,
                           OktaOAuth2Properties oktaOAuth2Properties) {
 
-            NimbusJwtDecoderJwkSupport decoder = new NimbusJwtDecoderJwkSupport(oAuth2ResourceServerProperties.getJwt().getJwkSetUri());
-            decoder.setJwtValidator(TokenUtil.jwtValidator(oAuth2ResourceServerProperties.getJwt().getIssuerUri(), oktaOAuth2Properties.getAudience()));
-            decoder.setRestOperations(restOperations());
-            return decoder;
+        NimbusJwtDecoderJwkSupport decoder = new NimbusJwtDecoderJwkSupport(oAuth2ResourceServerProperties.getJwt().getJwkSetUri());
+        decoder.setJwtValidator(TokenUtil.jwtValidator(oAuth2ResourceServerProperties.getJwt().getIssuerUri(), oktaOAuth2Properties.getAudience()));
+        decoder.setRestOperations(restOperations());
+        return decoder;
     }
 
+
     private RestOperations restOperations() {
         RestTemplate restTemplate = new RestTemplate();
         restTemplate.getInterceptors().add(new UserAgentRequestInterceptor());
         return restTemplate;
     }
+
+    @Bean
+    @Conditional(OktaOpaqueTokenIntrospectConditional.class)
+    OpaqueTokenIntrospector opaqueTokenIntrospector(OktaOAuth2Properties oktaOAuth2Properties,
+                                                    OAuth2ResourceServerProperties oAuth2ResourceServerProperties) {
+
+        OpaqueTokenIntrospector delegate = new NimbusOpaqueTokenIntrospector(
+            oAuth2ResourceServerProperties.getOpaquetoken().getIntrospectionUri(),
+            oAuth2ResourceServerProperties.getOpaquetoken().getClientId(),
+            oAuth2ResourceServerProperties.getOpaquetoken().getClientSecret());
```

Comment:
Can you set the RestTemplate here?

Reply:
Or is this object used for WebFlux too?

Reply:
Actually, there is no real need for us to set the `RestTemplate` here (but we could, if we'd like that customization here). `NimbusOpaqueTokenIntrospector` will new a `RestTemplate` for us if we don't set it here. This is being used for Webflux too.

Reply:
As discussed earlier today, pushed an update that would add `UserAgentRequestInterceptor` header to the `RestTemplate` injected by `NimbusOpaqueTokenIntrospector`.

Reply:
Are you still switching to the other constructor?

Reply:
If I switch to the other constructor (`NimbusOpaqueTokenIntrospector(String introspectionUri, RestOperations restOperations)`, I would still have to find a way to inject client-id and client-secret and then stub a `        restTemplate.getInterceptors().add(new BasicAuthenticationInterceptor(clientId, clientSecret))` in. Either way, it should be fine and I chose the other way.

Reply:
updated now.